packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -658,71 +665,135 @@ class ScrollNotification extends Notification {
   final ScrollableState scrollable;
 }
 
-/// A simple scrollable widget that has a single child. Use this widget if
+/// A simple scrolling widget that has a single child. Use this widget if
 /// you are not worried about offscreen widgets consuming resources.
-class ScrollableViewport extends Scrollable {
+class ScrollableViewport extends StatefulWidget {
   ScrollableViewport({
     Key key,
-    double initialScrollOffset,
-    Axis scrollDirection: Axis.vertical,
-    ViewportAnchor scrollAnchor: ViewportAnchor.start,
-    ScrollListener onScrollStart,
-    ScrollListener onScroll,
-    ScrollListener onScrollEnd,
+    this.initialScrollOffset,
+    this.scrollDirection: Axis.vertical,
+    this.scrollAnchor: ViewportAnchor.start,
+    this.onScrollStart,
+    this.onScroll,
+    this.onScrollEnd,
+    this.snapOffsetCallback,
+    this.scrollableKey,
     this.child
-  }) : super(
-    key: key,
-    scrollDirection: scrollDirection,
-    scrollAnchor: scrollAnchor,
-    initialScrollOffset: initialScrollOffset,
-    onScrollStart: onScrollStart,
-    onScroll: onScroll,
-    onScrollEnd: onScrollEnd
-  );
-
-  /// The widget below this widget in the tree.
-  final Widget child;
+  }) : super(key: key);
 
-  @override
-  ScrollableState createState() => new _ScrollableViewportState();
-}
+  /// The scroll offset this widget should use when first created.
+  final double initialScrollOffset;
 
-class _ScrollableViewportState extends ScrollableState<ScrollableViewport> {
-  @override
-  OverscrollWhenScrollableBehavior createScrollBehavior() => new OverscrollWhenScrollableBehavior();
+  /// The axis along which this widget should scroll.
+  final Axis scrollDirection;
+
+  /// Whether to place first child at the start of the container or
+  /// the last child at the end of the container, when the scrollable
+  /// has not been scrolled and has no initial scroll offset.
+  ///
+  /// For example, if the [scrollDirection] is [Axis.vertical] and
+  /// there are enough items to overflow the container, then
+  /// [ViewportAnchor.start] means that the top of the first item
+  /// should be aligned with the top of the scrollable with the last
+  /// item below the bottom, and [ViewportAnchor.end] means the bottom
+  /// of the last item should be aligned with the bottom of the
+  /// scrollable, with the first item above the top.
+  ///
+  /// This also affects whether, when an item is added or removed, the
+  /// displacement will be towards the first item or the last item.
+  /// Continuing the earlier example, if a new item is inserted in the
+  /// middle of the list, in the [ViewportAnchor.start] case the items
+  /// after it (with greater indices, down to the item with the
+  /// highest index) will be pushed down, while in the
+  /// [ViewportAnchor.end] case the items before it (with lower
+  /// indices, up to the item with the index 0) will be pushed up.
+  ///
+  /// Subclasses may ignore this value if, for instance, they do not
```

Comment:
there are three places where you now have this paragraph but as far as I can tell, none of the three are places where talking about subclasses makes sense.